### ISSUE TITLE
Move the CG method from extras module to a class method

### ISSUE THREAD

**Issue by chaithyagr** [mind-inria/mri-nufft#249]
On discussion, we realized it makes more sense to just add the CG algorithm as a class method allowing for easier user integtration. Additionally @alineyyy observed some robustness issues which need to be fixed.

What we expect would the users to do as compared to [current example](https://mind-inria.github.io/mri-nufft/generated/autoexamples/GPU/example_cg.html):

```python
fourier_op = get_operator("<backend>")(kspace_loc, img_size)
recon_cg, loss = fourier_op.cg(kspace_data, x_init, num_iterations, compute_loss=True) 
```
We can return the loss only when `compute_loss` is True (by default it is False). @paquiteau do you agree with this proposal?

 - [ ] Change the codebase and fix robustness issues as observed by @alineyyy 
 - [ ] Update the example to use the new method
 - [ ] Also the [current example](https://mind-inria.github.io/mri-nufft/generated/autoexamples/GPU/example_cg.html) needs to plot the k-space data from reconstructions and the acquired data and showcase the loss curve, to be really perfect.
 - [ ] Update the tests, add if they are missing.

**Comment by paquiteau** [comment]
compute_loss option  is fine by me (especially if its save computations when set to false)

Just to be picky: it will not be a class method, but just a method (you need to have an instantiated object to use it). 

It may also be nice to name it `pseudo_inverse` or `dagger_op` (à la deepinv)
